addons/mail/static/src/components/attachment_box/attachment_box.js

```diff
@@ -15,6 +17,8 @@ export class AttachmentBox extends Component {
     constructor(...args) {
         super(...args);
         this.isDropZoneVisible = useDragVisibleDropZone();
+        useComponentToModel({ fieldName: 'componentAttachementBox', modelName: 'mail.thread', propNameAsRecordLocalId: 'threadLocalId' });
+        useRefToModel({ fieldName: 'fileUploaderRef', modelName: 'mail.thread', propNameAsRecordLocalId: 'threadLocalId', refName: 'fileUploader' });
```

Comment:
For this PR I'm thinking the correct model is not thread but either thread view or probably even better chatter. Because thread record is "unique" whereas there could hypothetically be several attachment box, and then they would conflict with each other on thread.
(At least until we have 1 component = 1 record, then it would be attachment box itself of course.)